tools/prepare-release.js

```diff
@@ -0,0 +1,305 @@
+    { encoding: "utf8" });
+
+  // Clean up tmp folder
+  try {
+    rimraf.sync(tmpFolder);
+  }
+  catch {
+  }
+
+  // On Windows, the "echo -n" command actually outputs "-n"
+  // Also replace tmp folder name in diff to make it possible to compare diff
+  // with diff that may already exist in a pending pre-release PR.
+  // Note the diff command doubles backslashes, which also need to be doubled
+  // in the string passed to RegExp for them not to be interpreted (In practice,
+  // these backslashes also only appear on Windows)
+  diff = diff
+    .replace(/\-n\s*$/, "")
+    .replace(new RegExp(tmpFolder.replace(/\\/g, "\\\\\\\\"), "g"), "webref");
+
+  return diff;
+}
+
+
+/**
+ * Create or update pre-release pull request
+ *
+ * @function
+ * @param {String} type Package name. One of "css" or "idl"
+ */
+async function prepareRelease(type) {
+  // Compute a reasonably unique ID
+  const uid = (new Date()).toISOString().replace(/[\-T:\.Z]/g, "");
+
+  console.log("Look for a pending pre-release PR");
+  const searchResponse = await octokit.search.issuesAndPullRequests({
+    q: `repo:${owner}/${repo} type:pr state:open 📦 Release @webref/${type}@ in:title`
+  });
+  const found = searchResponse?.data?.items?.[0];
+
+  const pendingPRResponse = found ?
+    await octokit.pulls.get({
+      owner, repo,
+      pull_number: found.number
+    }) :
+    null;
+  const pendingPR = pendingPRResponse?.data;
+  console.log(pendingPR ?
+    `- Found pending pre-release PR: ${pendingPR.title} (${pendingPR.number})` :
+    "- No pending pre-release PR");
+
+  console.log();
+  console.log("Compute diff between package and repo contents");
+  const diff = computeDiff(type);
+  console.log(`- Diff length: ${diff?.length}`);
+  if (!diff) {
+    if (pendingPR) {
+      console.log("- No release needed, close pending pre-release PR");
+      await octokit.pulls.update({
+        owner, repo, pull_number: pendingPR.number,
+        body: "This pull request is no longer needed. No more diff to release.",
+        state: "closed"
+      });
+    }
+
+    console.log("- No diff found, return");
+    return;
+  }
+
+  console.log();
+  console.log("Extract and bump version number");
+  const packageFile = require(`../packages/${type}/package.json`);
+  const version = packageFile.version;
+  const bumpedVersion = version
+    .split(".")
+    .map((nb, idx) => parseInt(nb, 10) + ((idx === 2) ? 1 : 0))
+    .join(".");
+  packageFile.version = bumpedVersion;
+  const bumpedPackageFileContents = btoa(JSON.stringify(packageFile, null, 2));
+  console.log(`- Version to release: ${version}`);
+  console.log(`- Bumped version: ${bumpedVersion}`);
+
+  console.log();
+  console.log("Get latest commit on default branch");
+  const latestCommitResponse = await octokit.repos.getCommit({
+    owner, repo, ref: "HEAD"
+  });
+  const latestCommitSha = latestCommitResponse?.data?.sha;
```

Comment:
Using `?.` but still not checking if the result is null makes the code look undecided about the possibility of not finding the latest SHA.